File: IntegrationTests/Parser/TypeParsing/FragmentUsageTests/ImplementingClassWithInvalidFragmentUsage.cs
using InjectionCop.Fragment;

namespace InjectionCop.IntegrationTests.Parser.TypeParsing.FragmentUsageTests
{
  public class ImplementingClassWithInvalidFragmentUsage : IBaseInterface
  {
    public void Foo ([Fragment("InvalidFragmentType")]int parameter1, string parameter2)
    {
    }
  }
}

File: IntegrationTests.VB/Parser/TypeParsing/FragmentUsageTests/ImplementingClassWithInvalidFragmentUsage.vb
Imports InjectionCop.Fragment
Imports System

Namespace Parser.TypeParsing.FragmentUsageTests
	Public Class ImplementingClassWithInvalidFragmentUsage
		Implements IBaseInterface

    Public Sub Foo(<Fragment("InvalidFragmentType")> parameter1 As Integer, parameter2 As String) Implements IBaseInterface.Foo
    End Sub
	End Class
End Namespace
